pkg/router/template/router.go

```diff
@@ -398,6 +399,12 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 	config := ServiceAliasConfig{
 		Host: host,
 		Path: route.Spec.Path,
+		Hash: backendKey,
+	}
+
+	// Possibly hash the name for later use in obsuring cookie ID
+	if os.Getenv("COOKIE_PREFIX_MAIN") + os.Getenv("COOKIE_PREFIX_EDGE") + os.Getenv("COOKIE_PREFIX_REENCRYPT") != "" {
```

Comment:
Given that the environment won't change once the router has been started, is there a reason to check whether a hash is required on every route addition rather than at startup?

I'm also not why it would be preferable to concatenate the values rather than checking them individually.  Route addition isn't exactly a hot spot, though.